kamon-kafka-clients/src/main/scala/kamon/kafka/client/instrumentation/RecordProcessor.scala

```diff
@@ -0,0 +1,62 @@
+package kamon.kafka.client.instrumentation
+
+import java.time.Instant
+
+import kamon.Kamon
+import kamon.context.Context
+import kamon.kafka.client.Client
+import kamon.trace.Span
+import org.apache.kafka.clients.consumer.ConsumerRecords
+
+import scala.collection.mutable
+
+object RecordProcessor {
+
+  import scala.collection.JavaConverters._
+
+  def process[V, K](startTime: Instant, clientId: String, groupId: String, records: ConsumerRecords[K, V]): ConsumerRecords[K, V] = {
+    if (!records.isEmpty) {
+
+      val consumerSpansForTopic = new mutable.LinkedHashMap[String, Span]()
+
+      records.partitions().asScala.foreach(partition => {
+        val topic = partition.topic
+
+        records.records(partition).asScala.foreach(record => {
+          val header = Option(record.headers.lastHeader("kamon-context"))
+
+          val sendingContext = header.map { h =>
+            ContextSerializationHelper.fromByteArray(h.value())
+          }.getOrElse(Context.Empty)
+
+          val span = consumerSpansForTopic.getOrElseUpdate(topic, {
```

Comment:
Hey, i was wondering about poll-spans cardinality here. 
This way it will produce one span per consumed topic within a poll, is this intentional and whats the reasoning behind it?

Seams it could go two ways:
- One span per poll, in which case record specific tags dont makes sense (partition, key, offset). Any further streaming spans should link to poll span due to potentially large number of child spans (default `max.poll.records` is 500?)
- One span per polled record, can be used as a root for all streaming stages spans. This is aligned with what is done on producing side instrumentation which tracks individual record `send()`s and ingores batching.

Reply:
@mladens Thanks for your interest!
The current implementation creates one span per record. The main motivations for this are:
- propagate the originating traceId per record in order to trace the "business flow"
- create a parent-child relationships between the span of the published record and the subsequent spans of further processing, either in streams or an Akka-based infrastructure.

Reply:
Hi @mladens
thanks a lot for your feedback and for pointing out this span inconsistency!

I've refactored that part and now the following spans are created:
- one span for the actual `poll` operation: using its "own" (existing) traceId context this allows tracing of individual poll operations and how many records they actually polled.
- one span for each record polled: using the record's send record as parent span or linked span if `follow-strategy = false` this enable tracing of the business-related flow of the application.

This way it becomes also quite visible how Kafka consumer settings like `autoCommit` can influence the consumer. 

Here are two examples on how the instrumentation now looks like:

**Simple publish/subscribe with two messages (autoCommit=true)**
![client-autoCommit](https://user-images.githubusercontent.com/6660911/66701070-319e1900-ecf8-11e9-88fd-fb4eb1396c4a.png)

**Simple publish/subscribe with two messages (autoCommit=false)**
![client-noAutoCommit](https://user-images.githubusercontent.com/6660911/66701059-03203e00-ecf8-11e9-9ee3-1128255a4e12.png)

- solid edges: parent/child spans 
- dashed edges: linked spans

In the second example was the second record not commit fast enough so that it popped up in the next poll operation. It will not be forwarded to the application since it is still "in flight" on the client side.